File: 2017-02-02/W4D4-Archiving/W4D4-Archiving/ViewController.m
//
//  ViewController.m
//  W4D4-Archiving
//
//  Created by Jason Liang on 2/2/17.
//  Copyright © 2017 Jason Liang. All rights reserved.
//

#import "ViewController.h"
#import "Animal.h"
#import "SimpleObject.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextField *textField;

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
//  NSUserDefaults *ud = [NSUserDefaults standardUserDefaults];
//  SimpleObject *simpleObject = [SimpleObject new];
//  [ud setObject:simpleObject forKey:@"k"];
}

// Archiving Action
// Object -> NSData -> UserDefault
- (IBAction)archiveTapped:(id)sender {
  NSString *text = self.textField.text;
  Animal *animal = [[Animal alloc] init];
  animal.kind = text;
  NSData *animalData = [NSKeyedArchiver archivedDataWithRootObject:animal];
  NSUserDefaults *ud = [NSUserDefaults standardUserDefaults];
  [ud setObject:animalData forKey:@"kAnimalData"];
}

// Unarchiving Action
// UserDefault -> NSData -> Object
- (IBAction)unarchiveTapped:(id)sender {
  NSUserDefaults *ud = [NSUserDefaults standardUserDefaults];
  NSData *animalData = [ud objectForKey:@"kAnimalData"];
  Animal *animal = [NSKeyedUnarchiver unarchiveObjectWithData:animalData];
  self.textField.text = animal.kind;
}

// Store simple info directly into User Defaults.
- (IBAction)archiveTapped2:(id)sender {
  NSUserDefaults *ud = [NSUserDefaults standardUserDefaults];
  [ud setObject:_textField.text forKey:@"kTFText"];
}

- (IBAction)unarchiveTapped2:(id)sender {
  NSUserDefaults *ud = [NSUserDefaults standardUserDefaults];
  NSString *text = [ud objectForKey:@"kTFText"];
  self.textField.text = text;
}

@end

File: 2018-03-08/W8D4-Firebase/W8D4-Firebase/ViewController.swift
//
//  ViewController.swift
//  W8D4-Firebase
//
//  Created by Jason Liang on 3/8/18.
//  Copyright © 2018 jasonliang. All rights reserved.
//

import UIKit
import FirebaseDatabase

class ViewController: UIViewController {

    var ref : DatabaseReference!
    
    @IBOutlet weak var textField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Creating database reference
        ref = Database.database().reference()
    }
    
    @IBAction func addEntryTapped() {
        if let text = textField.text,
            text.count > 0 {
            addEntry(message: text)
        }
    }
    
    func addEntry(message:String) {
        let generatedRef = ref.child("entries").childByAutoId()
        generatedRef.description()
        generatedRef.setValue(message)
    }

}

